### PR TITLE
fix lnurl when callback already contains querystrings.

### PR DESCRIPTION
**Pull request by fiatjaf**
Fixes https://github.com/ZeusLN/zeus/issues/401.

This pull request is categorized as a:

- [ ] New feature
- [x] Bug fix
- [ ] Code refactor
- [ ] Configuration change
- [ ] Locales update
- [ ] Other

## Checklist
- [ ] I’ve run `npm run tsc` and make sure my code didn’t produce errors 
- [x] I’ve run `npm run prettier` and formatted my code correctly

## Testing

If you added new functionality or fixed a bug, did you add new unit tests?

- [x] No, I’m a fool
- [ ] Yes
- [ ] N/A

I have tested this PR on the following platforms (please specify OS version and phone model/VM):

- [ ] Android
- [ ] iOS

I have tested this PR with the following types of nodes (please specify node version and API version where appropriate):

- [ ] LND
- [ ] c-lightning-REST
- [ ] spark
- [ ] Eclair
- [ ] LndHub

### Locales
- [ ] I’ve added new locale text that requires translations
- [ ] I’m aware that new translations should be made on the Zeus [Transfix page](https://www.transifex.com/ZeusLN/zeus/) and not directly to this repo

### Third Party Dependencies and Packages

- [ ] Contributors will need to run `npm install` after this PR is merged in
- [ ] 3rd party dependencies have been modified:
    * verify that `package.json` and `package-lock.json` have been properly updated
    * verify that dependencies are installed for both iOS and Android platforms

### Other:

- [ ] Changes were made that require an update to the README
- [ ] Changes were made that require an update to onboarding

### CODE REVIEW

File: stores/InvoicesStore.ts [hunk 1]
```diff
@@ -118,11 +120,15 @@ export default class InvoicesStore {
                 this.payment_request = invoice.getPaymentRequest;
                 this.creatingInvoice = false;
 
+                const u = url.parse(lnurl.callback);
+                const qs = querystring.parse(u.query);
+                qs.k1 = lnurl.k1;
+                qs.pr = this.payment_request;
+                u.search = querystring.stringify(qs);
+                u.query = querystring.stringify(qs);
+
```
Comment (kaloudis) [inline stores/InvoicesStore.ts:129]: These lines should be in the if block below as they're only applicable if lnurl is defined.

File: stores/InvoicesStore.ts [hunk 2]
```diff
@@ -84,7 +86,7 @@ export default class InvoicesStore {
             .then((data: any) => {
                 this.invoices = data.payments || data.invoices || data;
                 this.invoices = this.invoices.map(
-                    invoice => new Invoice(invoice)
+                    (invoice) => new Invoice(invoice)
```
Comment (kaloudis) [inline stores/InvoicesStore.ts:89]: you can remove the parentheses around invoice unless you want to give it a type

**Review by kaloudis — CHANGES_REQUESTED** [review]

File: stores/InvoicesStore.ts (hunk 1 above)
Comment (fiatjaf) [reply stores/InvoicesStore.ts:129]: Oops.

File: stores/InvoicesStore.ts (hunk 2 above)
Comment (fiatjaf) [reply stores/InvoicesStore.ts:89]: They were applied by Prettier automatically!

**Comment by fiatjaf** [comment]
Sorry for the stupid mistakes. Fixed them now.

**Review by ZeusLN — APPROVED** [review]

**Pull request merged**